R/download.r

```diff
@@ -37,27 +38,50 @@
 download <- function(url, ...) {
   # First, check protocol. If http or https, check platform:
   if (grepl('^https?://', url)) {
-
-    # If Windows, call setInternet2, then use download.file with defaults.
+    
+    # Check whether we are running R 3.2
+    isR32 <- getRversion() >= "3.2"
+    
+    # Windows
     if (.Platform$OS.type == "windows") {
-
-      # If we directly use setInternet2, R CMD CHECK gives a Note on Mac/Linux
-      seti2 <- `::`(utils, 'setInternet2')
-
-      # Store initial settings, and restore on exit
-      internet2_start <- seti2(NA)
-      on.exit(suppressWarnings(seti2(internet2_start)))
-
-      # Needed for https. Will get warning if setInternet2(FALSE) already run
-      # and internet routines are used. But the warnings don't seem to matter.
-      suppressWarnings(seti2(TRUE))
-      download.file(url, ...)
-
+      
+      if (isR32) {
+        method <- "wininet"
```

Comment:
No, there is no capabilities for "wininet" it's always there on R for
Windows 3.2

On Tue, Jul 7, 2015 at 2:49 PM, Joe Cheng <notifications@github.com> wrote:

> In R/download.r
> <https://github.com/wch/downloader/pull/5#discussion_r34075903>:
>
> > -
> > -      # If we directly use setInternet2, R CMD CHECK gives a Note on Mac/Linux
> > -      seti2 <- `::`(utils, 'setInternet2')
> > -
> > -      # Store initial settings, and restore on exit
> > -      internet2_start <- seti2(NA)
> > -      on.exit(suppressWarnings(seti2(internet2_start)))
> > -
> > -      # Needed for https. Will get warning if setInternet2(FALSE) already run
> > -      # and internet routines are used. But the warnings don't seem to matter.
> > -      suppressWarnings(seti2(TRUE))
> > -      download.file(url, ...)
> > -
> > +
> > +      if (isR32) {
> > +        method <- "wininet"
>
> Does wininet need to be tested for using capabilities() as well?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wch/downloader/pull/5/files#r34075903>.
>